src/pool/srv_target.c

```diff
@@ -928,6 +928,13 @@ ds_pool_tgt_map_update(struct ds_pool *pool, struct pool_buf *buf,
 				D_GOTO(out, rc);
```

Comment:
Big chunk of code in this function is duplicated, it would be great if we can re-organize it a bit to remove the dup code.